sgt-back-end/server.js

```diff
@@ -0,0 +1,150 @@
+import express from 'express';
+import pg from 'pg';
+const app = express();
+const port = 8080;
+
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+// GET / api / grades returns all rows from the "grades" table.The client should receive an array of objects.If there happens to be no rows, an empty array is ok.
+app.get('/api/grades', async (req, res) => {
+  try {
+    // sql query
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    const grade = result.rows;
+    res.status(200).json(grade);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occured.' });
+  }
+});
+// POST /api/grades, insert new grade and return created grade. should return an object
+
+app.post('/api/grades', async (req, res) => {
+  try {
+    // check for missing values
+    const { name, course } = req.body;
+    const score = Number(req.body.score);
+    // check values
+    if (!name) {
+      res.status(400).json({ error: 'name must be provided' });
+    } else if (!course) {
+      res.status(400).json({ error: 'course must be provided' });
+    } else if (!score || score <= 0 || score > 100) {
+      res.status(400).json({ error: 'score must be a positive integer' });
```

Comment:
A `score` of `0` should not cause a `400 Bad Request` response:

---

![image](https://github.com/rachel-lyy/m2-0323-code-solutions/assets/16143771/353a4e94-0108-4498-8554-838f4c4b8e86)